src/types/__test__/1/collection.test.ts

```diff
@@ -785,9 +828,73 @@ describe('update pinned', () => {
   })
 })
 
+describe('get collection by article id', () => {
+  let server: Awaited<ReturnType<typeof testClient>>
+  let collectionId: string
+
+  beforeAll(async () => {
+    server = await testClient({ isAuth: true, connections })
+    const { data } = await server.executeOperation({
+      query: PUT_COLLECTION,
+      variables: { input: { title: 'test' } },
+    })
+    collectionId = data?.putCollection?.id
+    await server.executeOperation({
+      query: ADD_COLLECTIONS_ARTICLES,
+      variables: {
+        input: {
+          collections: [collectionId],
+          articles: [articleGlobalId1, articleGlobalId4],
+        },
+      },
+    })
+  })
+  test('get collection by article id', async () => {
+    const { data } = await server.executeOperation({
+      query: GET_COLLECTION_BY_ARTICLES,
+      variables: {
+        input1: { id: collectionId },
+        input2: { articleId: articleGlobalId1 },
+      },
+    })
+    expect(data?.node?.id).toBe(collectionId)
+    expect(data?.node?.articles?.edges?.[0]?.node?.id).toBe(articleGlobalId4)
```

Comment:
if the full list is `[articleGlobalId4, articleGlobalId1]`, we expect the return here should be `[articleGlobalId1]` (start from given `articleId`).

Reply:
The original design of the returned page was like this:
```ts
const articles = [1, 2, 3, 4]

// if querying with id=3, take = 2
const result = [3, 4]
// if querying with id=4, take =2
const result = [3, 4] // the same
```
Because if every time if we query the article id, I don't want to break the original pages and fragment them. let me know what you think.